### DISCUSSION TITLE
Lesson 7: testFailsWithoutEnoughEth issue

### DISCUSSION THREAD

**Discussion by Creol1** [Cyfrin/foundry-full-course-cu#1020] [Q&A] [answered]
Hi All,

when trying to test testFailsWithoutEnoughEth i recive an error (on picture). this happens on local and sepolia tests
<img width="609" alt="Screenshot 2023-10-16 at 21 28 20" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/49288870/62573bf7-7baa-4ac6-abe4-6b11ef67fe35">

**Comment by 0xusmanf** [comment]
Please provide your test function.

**Reply by Creol1** [reply]
here:
`function testFailsWithoutEnoughEth() public {
        vm.expectRevert();
        fundMe.fund();
    }`

**Reply by 0xusmanf** [reply]
Please provide the error log after running the following command `forge test --mt testFailsWithoutEnoughEth -vvvvv`

**Reply by Creol1** [reply]
Attached
<img width="677" alt="Screenshot 2023-10-16 at 21 42 21" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/49288870/2dd2a2f0-cddf-4950-b92d-82d3ae1a1f05">

**Reply by 0xusmanf** [reply]
There is nothing wrong with your test your test is working as expected but why you are getting an Assertion failed error is weird.

Can you upload your full repo on GitHub and give the link, so I can recreate it?

**Reply by 0xusmanf** [reply]
If you don't know how to upload to Github, then just provide all the files here.

**Reply by Creol1** [reply]
here is a link: https://github.com/Creol1/foundry-fund-me-f23

**Reply by 0xusmanf** [reply]
Please recheck the link, you have provided the link to this discussion.

**Reply by Creol1** [reply]
sorry, updated the link

**Reply by 0xusmanf** [reply, marked as answer]
This is super weird but I changed the function name to `testFundFailsWithoutEnoughEth` and it passed. So I suggest you change the function name too.

**Reply by Creol1** [reply]
that is super strange! some coding magic, but it really works now. thanks a lot

**Reply by 0xusmanf** [reply]
You are welcome, always happy to help.
